File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int p1=0;
        int p2=0;
        
        for(int i = 0; i < goal.length; i++) {
            boolean flag = true;
            
            if(p1 != cards1.length && goal[i].equals(cards1[p1])) {
                flag = false;
                p1++;
            }
            
            if(p2 != cards2.length && goal[i].equals(cards2[p2])) {
                flag = false;
                p2++;
            }
            
            if(flag) return "No";
        }
        return "Yes";
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    p1 = 0
    p2 = 0
    
    for s in goal:
        if p1 < len(cards1) and cards1[p1] == s:
            p1 += 1
            continue
    
        if p2 < len(cards2) and cards2[p2] == s:
            p2 += 1
            continue
        
        return "No"
    return "Yes"